plugins/processors/topk/README.md

```diff
@@ -0,0 +1,25 @@
+# TopK Processor Plugin
+
+The TopK processor plugin is a filter that keeps only the top k values of a given series. It can be tweaked to do its top k computation over a period of time, so spikes can be smoothed out.
+
+### Configuration:
+
+```toml
+[[processors.topk]]
+  period = 10                  # How many seconds between aggregations. Default: 10
+  k = 10                       # How many top metrics to return. Default: 10
+  metric = "mymetric"          # Which metrics to consume. Supports regular expressions. No default. Mandatory
+  tags = {"tag_name"="tag_value"}  # Map of tags regexes to match against. Default: "{}" (match all)
+  fields = ["memory_rss"]      # Over which fields are the top k are calculated. Default: "value"
+  aggregation = "avg"          # What aggregation to use. Default: "avg". Options: sum, avg, min, max
+  group_by = ["process_name"]  # Over which tags should the aggregation be done. Default: []
+  group_by_metric_name = false # Wheter or not to also group by metric name
```

Comment:
Will it work if we always group by series key?  The series key is `measurment,tagkey=tagvalue,tagkey2=tagvalue2` and you can use `metric.HashID()` to get an opaque series identifier.

Reply:
I don't quite follow you here. It will work if you always group by metric name and all the tags. Is that what you want to know?

From what I understand metric.HashID() returns some hash over the series name and the tags. That is useful, but for this use case, it would prevent me from grouping over only tags, or only over a subset of tags.

Reply:
Can you give an example of when grouping over a subset of tags is useful?

Reply:
You could have metrics of CPU from several datacenters, but want to group only by process name, so as to get a picture of the most intensive processes across all data-centers.

A more concrete example, the procstat module puts the process_name in a tag, but you can optinally put the PID in a tag. If you put the PID in a tag, grouping across all tags will agreggate invocations of processes, which might not be what you need as it fails to agreggate across process restarts.

Reply:
In a more general note, I think grouping across a subset of tags gives the plugin much more expressive power that has an excellent potential of addressing some a lot of potential user needs.

Reply:
:+1: